service/src/main/java/bio/terra/catalog/service/DatasetService.java

```diff
@@ -72,8 +73,8 @@ private void collectDatarepoDatasets(
     // Merge the permission (role) data into the metadata results.
     for (Dataset dataset : datasets) {
       ArrayNode roles = objectMapper.createArrayNode();
-      for (String role : roleMap.get(dataset.storageSourceId())) {
-        roles.add(TextNode.valueOf(role));
+      for (DatasetAccessLevel role : roleMap.get(dataset.storageSourceId())) {
+        roles.add(TextNode.valueOf(role.toString()));
```

Comment:
Very minor but I usually prefer `String.valueOf()` as it works on all java types, unlike `toString()` which only works on Objects.
```suggestion
        roles.add(TextNode.valueOf(String.valueOf(role)));
```